src/python_bindings/bindings.cpp

```diff
@@ -155,11 +261,115 @@ class PyHNSWLibIndex : public PyVecSimIndex {
         auto *hnsw = reinterpret_cast<HNSWIndex<float, float> *>(index.get());
         hnsw->saveIndex(location);
     }
+    py::object searchKnnParallel(const py::object &input, size_t k, VecSimQueryParams *query_params,
+                                 int n_threads) {
+
+        py::array queries(input);
+        if (queries.ndim() != 2) {
+            throw std::runtime_error("Input queries array must be 2D array");
+        }
+        size_t n_queries = queries.shape(0);
+        std::function<VecSimQueryResult_List(const char *, size_t, VecSimQueryParams *)>
+            searchKnnWrapper([this](const char *query_, size_t k_,
+                                    VecSimQueryParams *query_params_) -> VecSimQueryResult_List {
+                return this->searchKnnInternal(query_, k_, query_params_);
+            });
+        VecSimQueryResult_List results[n_queries];
+        runParallelQueries<size_t>(queries, n_queries, k, query_params, n_threads, searchKnnWrapper,
+                                   results);
+        return wrap_results(results, k, n_queries);
+    }
+    py::object searchRangeParallel(const py::object &input, double radius,
+                                   VecSimQueryParams *query_params, int n_threads) {
+        py::array queries(input);
+        if (queries.ndim() != 2) {
+            throw std::runtime_error("Input queries array must be 2D array");
+        }
+        size_t n_queries = queries.shape(0);
+        std::function<VecSimQueryResult_List(const char *, double, VecSimQueryParams *)>
+            searchRangeWrapper([this](const char *query_, double radius_,
+                                      VecSimQueryParams *query_params_) -> VecSimQueryResult_List {
+                return this->searchRangeInternal(query_, radius_, query_params_);
+            });
+        VecSimQueryResult_List results[n_queries];
+        runParallelQueries<double>(queries, n_queries, radius, query_params, n_threads,
+                                   searchRangeWrapper, results);
+        size_t max_results_num = 1;
```

Comment:
I did it to avoid the case of creating a matrix with `0` as one of its axes in `wrap_results`, which may complicate things in the corner cases where all queries returned 0 results. It doesn't really matter, we use padding in the redundant places anyway...